### DISCUSSION TITLE
changing the icons

### DISCUSSION THREAD

**Discussion by DimaM-BeStreet** [add2cal/add-to-calendar-button#95] [Ideas / Requests]
Hey, love what you did, thank you!
one thing i'd love to have is the option to change the icons for the calendars, i have a problem with the colors of GCal, it does not fit into my color scheme in my app

(I mean, being able to use my own pre-made svg instead)

Thank you!
Dima

**Comment by jekuer** [comment]
Hi @DimaM-BeStreet ,

Thanks for your message.

We should not allow this to be configurable the way we do it for other things. Simply because there are underlying security risks for that.
Theoretically, you could simply exchange the icon in the source code and build it yourself. However, I see that this is not very convenient, especially when it comes to future updates.
Another idea: We could remove the icons as objects and rather define them as an attribute for a plain object - and than add an option to not load them. That approach would make it definable via css. You would need to then define them in your css somewhere.

Something like that.

Would that be an option for you?

**Reply by DimaM-BeStreet** [reply]
That would be great, i could use the colorless version of the icons that way

btw i am now using a simple filter: saturate(0%); css property to grayscale the icons, seems to be ok (but not great hehe)

If it's not too much trouble, can you explain the security risks?

**Reply by jekuer** [reply]
svg files are pure raw code. Loading them dynamically via JS opens potential code infusion risks. You can sanitize the input, but this makes the whole thing extremely bulky for the only gain of changing the icons in 1 out of 1000 cases.
The option to hide the hard coded ones and override it with basically other hard coded one makes it one step less of exchanging potentially malicious data.

The way to go is quite clear. The exact final implementation needs to be evaluated along the way. But we will get there :) .

**Reply by DimaM-BeStreet** [reply]
Thank you! I've learned something new today :)
